frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -462,9 +462,13 @@ fn decl_all_modules<'a>(
 		.filter(|n| **n != SYSTEM_MODULE_NAME)
 		.fold(TokenStream2::default(), |combined, name| quote!((#name, #combined)));
 
+	let all_modules_with_system = names.iter()
+		.fold(TokenStream2::default(), |combined, name| quote!((#name, #combined)));
+
 	quote!(
 		#types
 		type AllModules = ( #all_modules );
+		type AllModulesWithSystem = ( #all_modules_with_system );
```

Comment:
(sidenote: I think we should make those types public so that users can see from doc what are the generated types.)

Reply:
Hmm, what is the benefit for the user?

Reply:
indeed there isn't that much benefit, though it used as a generic of Executive, I tend to prefer that all type used by user goes into doc, anyway its fine.